src/extension/serializer.ts

```diff
@@ -26,12 +31,58 @@ const DEFAULT_LANG_ID = 'text'
 
 type ReadyPromise = Promise<void | Error>
 
-export abstract class SerializerBase implements NotebookSerializer {
-  protected readonly abstract ready: ReadyPromise
+export abstract class SerializerBase implements NotebookSerializer, Disposable {
+  protected abstract readonly ready: ReadyPromise
   protected readonly languages: Languages
+  protected disposables: Disposable[] = []
 
   constructor(protected context: ExtensionContext) {
     this.languages = Languages.fromContext(this.context)
+    this.disposables.push(
+      workspace.onDidChangeNotebookDocument(
+        this.handleNotebookChanged.bind(this)
+      )
+    )
+  }
+
+  public dispose() {
+    this.disposables.forEach(d => d.dispose())
+  }
+
+
+  /**
+   * Handle newly added cells (live edits) to have UUIDs
+   */
+  protected handleNotebookChanged(changes: NotebookDocumentChangeEvent) {
+    changes.contentChanges.forEach((contentChanges) => {
+      contentChanges.addedCells.forEach((cellAdded) => {
+        if (
+          cellAdded.kind !== NotebookCellKind.Code ||
+          cellAdded.metadata['runme.dev/uuid'] !== undefined
+        ) {
+          return
+        }
+
+        const notebookEdit = NotebookEdit.updateCellMetadata(
```

Comment:
Oh yeah, one more thing: We should make sure this `NotebookEdit` is not `Ctrl+z`'able, since that would make it possible to "undo" the operation of adding a uuid.

Reply:
It is not. I did look into this initial when I first started using this API.